src/components/ContentNode/SectionTitle.vue

```diff
@@ -0,0 +1,86 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <component
+   :id="!isTargetIDE ? id : null"
+   :is="tag"
+   class="section-title"
+  >
+    <a
+      v-if="anchor && !isTargetIDE"
+      :href="`#${anchor}`"
+      class="header-anchor"
+      aria-label="hidden"
+      @click="handleFocusAndScroll(anchor)"
+    >#</a>
+    <slot />
+  </component>
+</template>
+
+<script>
+import scrollToElement from 'docc-render/mixins/scrollToElement';
+
+export default {
+  name: 'SectionTitle',
+  mixins: [scrollToElement],
+  data() {
+    return {
+      id: null,
+    };
+  },
+  props: {
+    anchor: {
+      type: String,
+      required: false,
+    },
+    tag: {
+      type: String,
+      default: () => 'h2',
+    },
+  },
+  inject: {
+    isTargetIDE: {
+      default: () => false,
+    },
+  },
+  mounted() {
+    if (!this.anchor) return;
+    const element = document.getElementById(`${this.anchor}`);
+    // if there is no element in the document that has an id,
+    // add it to this component
+    if (!element) {
+      this.id = this.anchor;
+    }
+  },
+};
+</script>
+
+<style scoped lang="scss">
+@import 'docc-render/styles/_core.scss';
+
+.section-title:hover {
+  .header-anchor {
+    opacity: 1;
+  }
+}
+
+.header-anchor {
+  margin-left: -0.73em;
+  padding-right: 0.23em;
+  transition: opacity .25s;
```

Comment:
You may want to have someone from the design team review this new feature and test the experience in different browsers. The scrolling animation specifically (not this one) feels a little different in Safari but that may be a limitation of the JS scroll function itself.